### PR TITLE
Add deployments from rpm ostree status to system profile.

### PR DESCRIPTION
**Pull request by csams**
Signed-off-by: Christopher Sams <csams@redhat.com>

### CODE REVIEW

File: schemas/system_profile/v1.yaml [hunk 1]
```diff
@@ -131,6 +131,48 @@ $defs:
...
+    properties:
+      id:
+        description: ID of the deployment
+        example: "fedora-silverblue-63335a77f9853618ba1a5f139c5805e82176a2a040ef5e34d7402e12263af5bb.0"
+        type: string
+        minLength: 1
+        maxLength: 256
+      checksum:
+        description: The checksum / commit of the deployment
+        example: "63335a77f9853618ba1a5f139c5805e82176a2a040ef5e34d7402e12263af5bb"
```
Comment (ryandillinfelton) [inline schemas/system_profile/v1.yaml:147]: Is this hex? We could use regex here to insure that it's valid hex AND specifically 64 characters long.

File: tests/test_schema.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from openapi_spec_validator import validate_spec
+# from openapi_spec_validator import validate_spec
```
Comment (ryandillinfelton) [inline tests/test_schema.py:1]: if you don't need this could you delete the whole line please.

File: tests/utils/invalids.py [hunk 3]
```diff
@@ -81,4 +81,39 @@
     {"greenboot_status": "blue"},
     {"greenboot_status": "too long"},
     {"greenboot_fallback_detected": "x"},
+    {"rpm_ostree_deployments": [{  # checksum empty
```
Comment (ryandillinfelton) [inline tests/utils/invalids.py:84]: This testing cold be more thorough. You only verify the checksum validation here.

**Review by ryandillinfelton — CHANGES_REQUESTED** [review]
Requesting:

- [ ] Stricter checksum validation
- [ ] more invalid examples
- [ ] some cleanup

**Comment by gmcculloug** [comment]
@beav Can you please review the OSTree properties and structure here and verify it makes sense?

@csams Do you think we'll need to be able to search on any of these properties?

File: tests/test_schema.py (hunk 2 above)
Comment (csams) [reply tests/test_schema.py:1]: Sure, I'll delete it. Tests won't even run with the line, so I'm not sure how anyone is testing this repo with it as-is.

File: schemas/system_profile/v1.yaml (hunk 1 above)
Comment (csams) [reply schemas/system_profile/v1.yaml:147]: Sounds good.

File: tests/utils/invalids.py (hunk 3 above)
Comment (csams) [reply tests/utils/invalids.py:85]: done

**Comment by csams** [comment]
@gmcculloug We'll need to make the checksum searchable.

**Comment by beav** [comment]
> @gmcculloug We'll need to make the checksum searchable.

Do you just need "matches first part of string" or full substring?

Like if I had: `1234abc` and `34a777`, if I searched for `34a` would you expect one result or 2 results?

**Comment by csams** [comment]
@beav @gmcculloug We can defer the search capability for now if it can be added later.

**Review by ryandillinfelton — APPROVED** [review]
looks good now. Thanks Chris!

**Review by kruai — APPROVED** [review]
LGTM

**Pull request merged**